provisioning/miniprov/test/test_validator.py

```diff
@@ -0,0 +1,163 @@
+    },
+    "cluster_id": "e766bd52-c19c-45b6-9c91-663fd8203c2e",
+    "srvnode-1": {
+      "storage_set_id": 0,
+      "hostname": "ssc-vm-1623.colo.seagate.com",
+      "node_type": "VM",
+      "roles": [
+        "primary",
+        "openldap_server",
+        "kafka_server"
+      ],
+      "is_primary": "true",
+      "bmc": {
+      },
+      "network": {
+        "mgmt": {
+          "interfaces": [
+            "eth0"
+          ],
+        },
+        "data": {
+          "public_interfaces": [
+            "eth0"
+          ],
+          "private_interfaces": [
+            "eth0"
+          ],
+          "interface_type": "tcp",
+          "transport_type": "lnet",
+          "private_ip": "",
+          "roaming_ip": "127.0.0.1"
+        }
+      },
+      "storage": {
+        "enclosure_id": "enclosure-1",
+        "metadata_devices": [
+          "/dev/sdb"
+        ],
+        "data_devices": [
+          "/dev/sdd",
+          "/dev/sde",
+          "/dev/sdf",
+          "/dev/sdg"
+        ]
+      },
+      "s3_instances": 1
+    }
+  }}
+
+
+def update_machine(machine_id: str, hostname: str):
+    with open ("/tmp/test-conf-store.json", "w+") as conf_store:
+        json.dump(conf_store_data, conf_store)
+
+    with open("/tmp/test-conf-store.json", "r+") as jf:
+        data = json.load(jf)
+        new_machine = {"server_node": {
+                                       f'{machine_id}': {
+                                         "name": "srvnode-1",
+                                         "hostname": f'{hostname}',
+                                         "cluster_id": "e766bd52-c19c-45b6-9c91-663fd8203c2e",
+                                         "storage_set_id": 0
+                                       }
+                                      }
+                      }
+        data.update(new_machine)
+        data['cluster']['e766bd52-c19c-45b6-9c91-663fd8203c2e']['storage_set'][0]['server_nodes'] = [f'{machine_id}']
+        with open("/tmp/temp-test-conf-store.json", "w+") as ujf:
+            json.dump(data, ujf)
+
+class TestValidator(unittest.TestCase):
+    def test_is_cluster_first_node(self):
+        conf = ConfStoreProvider(URL)
+        hostname = socket.gethostname()
+        machine_id = conf.get_machine_id()
+        update_machine(machine_id, hostname)
+        validator = Validator(
+                        ConfStoreProvider("json:///tmp/temp-test-conf-store.json"))
+        assert validator.is_first_node_in_cluster()
+
+    def test_invalid_machine_id(self):
+        conf = ConfStoreProvider(URL)
+        hostname = 'invalid-hostname'
+        machine_id = conf.get_machine_id()
+        update_machine(machine_id, hostname)
+        validator = Validator(
+                        ConfStoreProvider("json:///tmp/temp-test-conf-store.json"))
+        try:
+            machine_id = validator._get_machine_id()
+        except RuntimeError:
+            return
```

Comment:
Unit tests should be fail-fast: if anything abnormal happens, shout loudly.

Why do we swallow this exception? Can't it hide a greater problem?

Reply:
BTW you probably should take a look at [`assertRaises`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises)

It will pass if and only if the given exception is raised. Otherwise it will fail the test.

Reply:
This is a negative test case and we are checking if the failure path is taken care of. So if the machine-id does not match the hostname (for which we have a handling in the code) we need to make sure that the code raise appropriate exception so that we do not go ahead and fail later. Thus we are putting some invalid values in the given conf-store and making sure that the main code raises the exception but test case itself must pass as our expectation matches.

Will look into assertRaises, thanks.

Reply:
Done.